db/views/reporting_overdue_patients_v01.sql

```diff
@@ -0,0 +1,194 @@
+WITH patients_with_appointments AS (
+    SELECT
+        DISTINCT ON (rps.patient_id, rps.month_date)
+        rps.month_date,
+        rps.patient_id,
+        rps.hypertension AS hypertension,
+        rps.diabetes AS diabetes,
+        rps.htn_care_state,
+        rps.month,
+        rps.quarter,
+        rps.year,
+        rps.month_string,
+        rps.quarter_string,
+        rps.assigned_facility_id,
+        rps.assigned_facility_slug,
+        rps.assigned_facility_region_id,
+        rps.assigned_block_slug,
+        rps.assigned_block_region_id,
+        rps.assigned_district_slug,
+        rps.assigned_district_region_id,
+        rps.assigned_state_slug,
+        rps.assigned_state_region_id,
+        rps.assigned_organization_slug,
+        rps.assigned_organization_region_id,
+        appointments.id AS previous_appointment_id,
+        appointments.device_created_at AS previous_appointment_date,
+        appointments.scheduled_date AS previous_appointment_schedule_date
```

Comment:
I tested this on BD prod, it took 44s vs 80s (for this subquery alone)
```
WITH latest_appts_pp_pm as (
  select distinct on (appointments.patient_id, date_trunc('month', appointments.device_created_at)) *, date_trunc('month', appointments.device_created_at) month_date
  from appointments
  order by patient_id, date_trunc('month', appointments.device_created_at) desc
),
  patients_with_appointments AS (
    SELECT
    ...
    FROM reporting_patient_states rps
    LEFT JOIN latest_appts_pp_pm ON latest_appts_pp_pm.patient_id = rps.patient_id
        AND latest_appts_pp_pm.month_date = rps.month_date - interval '1 month'
    WHERE rps.status <> 'dead'
	AND rps.month_date > NOW() - INTERVAL '24 months'
)
```

Reply:
The logic won't work because the latest appointment per month can be a month in the previous month also. This is a good optimization though, however this relative lookup makes it not suitable here